src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -481,8 +482,11 @@ private ByteBuffer current() throws IOException {
                     if (debug.on()) debug.log("Next Buffer");
                     currentBuffer = currentListItr.next();
                 } catch (InterruptedException ex) {
-                    close();
-                    throw new IOException("interrupted", ex);
+                    try {
+                        close();
+                    } catch (IOException ignored) {
+                    }
+                    throw new InterruptedIOException();
```

Comment:
Thanks for doing that. Maybe we should keep the caught InterruptedException as the cause of the new InterruptedIOException. I'd suggest to add a new utility method to the `...common.Utils` class for that. I mean - something like:
```
    InterruptedIOException Utils.toInterruptedIOException(InterruptedException ex);
```

Reply:
So  something in utils like `return new InterruptedIOException(String.valueOf(exxeption));` ?

Reply:
No something that would call `initCause(ex)` to set the root cause - as AFAICS there's no constructor that takes a root cause.

Reply:
ah ok, so create a new `InterruptedIOException` and set the initcause to be the `interruptedException`.
is there anything else that would need to be set?

Reply:
preserving the root cause is the main thing.